components/calendar/src/hebrew.rs

```diff
@@ -0,0 +1,1230 @@
+    /// assert_eq!(date_hebrew.month().ordinal, 4);
+    /// assert_eq!(date_hebrew.day_of_month().0, 25);
+    /// ```
+    pub fn try_new_civil_hebrew_date(
+        year: i32,
+        month: u8,
+        day: u8,
+    ) -> Result<Date<CivilHebrew>, CalendarError> {
+        ArithmeticDate::new_from_lunar_ordinals(year, month, day)
+            .map(CivilHebrewDateInner)
+            .map(|inner| Date::from_raw(inner, CivilHebrew))
+    }
+}
+
+impl DateTime<CivilHebrew> {
+    /// Construct a new CivilHebrew datetime from integers.
+    ///
+    /// ```rust
+    /// use icu::calendar::DateTime;
+    ///
+    /// let datetime_hebrew = DateTime::try_new_civil_hebrew_datetime(4201, 10, 11, 13, 1, 0)
+    ///     .expect("Failed to initialize CivilHebrew DateTime instance");
+    ///
+    /// assert_eq!(datetime_hebrew.date.year().number, 4201);
+    /// assert_eq!(datetime_hebrew.date.month().ordinal, 10);
+    /// assert_eq!(datetime_hebrew.date.day_of_month().0, 11);
+    /// assert_eq!(datetime_hebrew.time.hour.number(), 13);
+    /// assert_eq!(datetime_hebrew.time.minute.number(), 1);
+    /// assert_eq!(datetime_hebrew.time.second.number(), 0);
+    /// ```
+    pub fn try_new_civil_hebrew_datetime(
+        year: i32,
+        month: u8,
+        day: u8,
+        hour: u8,
+        minute: u8,
+        second: u8,
+    ) -> Result<DateTime<CivilHebrew>, CalendarError> {
+        Ok(DateTime {
+            date: Date::try_new_civil_hebrew_date(year, month, day)?,
+            time: types::Time::try_new(hour, minute, second, 0)?,
+        })
+    }
+}
+
+// BIBLICAL HEBREW CALENDAR FUNCTIONS
+
+impl BookHebrew {
+    // Moment of mean conjunction (New Moon) of h_month in BookHebrew
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2244
+    #[allow(dead_code)]
+    pub(crate) fn molad(book_year: i32, book_month: u8) -> Moment {
+        let y = if book_month < TISHRI {
+            book_year + 1
+        } else {
+            book_year
+        }; // Treat Nisan as start of year
+
+        let months_elapsed = (book_month as f64 - TISHRI as f64) // Months this year
+            + (libm::floor((235.0 * y as f64 - 234.0) / 19.0)); // Months until New Year.
+
+        Moment::new(
+            FIXED_HEBREW_EPOCH.to_f64_date() - (876.0 / 25920.0)
+                + months_elapsed * (29.0 + (1.0 / 2.0) + (793.0 / 25920.0)),
+        )
+    }
+
+    // ADAR = 12, ADARII = 13
+    // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2217
+    #[allow(dead_code)]
+    fn last_month_of_book_hebrew_year(book_year: i32) -> u8 {
+        if Self::is_hebrew_leap_year(book_year) {
+            ADARII
+        } else {
+            ADAR
+        }
+    }
+
+    // Sabbatical Year of the BookHebrew Calendar
+    // Lisp code refernence: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L2224
+    #[allow(dead_code)]
```

Comment:
Nit: Please remove all `#[allow(dead_code)]` markers that are no longer necessary. It is OK to keep ones on BookHebrew that mark functions that you're not using.